first-react-app/src/components/formComponents/SubmitForm.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from "react";
+import { FieldErrors, useForm } from "react-hook-form";
+
+function SubmitForm(props: { isFormCorrect: (arg0: string[]) => void }) {
+  const [popUpClass, setpopUpClass] = useState("error-hidden");
+  const [popUpText, setpopUpText] = useState(
+    "You have to correct errors in form first only then you can submit."
+  );
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+    formState: { isSubmitSuccessful },
+  } = useForm({
+    defaultValues: {
+      songName: "Best Song In The World",
+      date: "2023-01-01",
+      select: "1",
+      checkbox: false,
+      radio: null,
+      upload: null,
+    },
+  });
+
+  useEffect(() => {
+    if (isSubmitSuccessful) {
+      reset({
+        songName: "Best Song In The World",
+        date: "2023-01-01",
+        select: "1",
+        checkbox: false,
+        radio: null,
+        upload: null,
+      });
+    }
+  }, [reset, isSubmitSuccessful]);
+
+  function handleErrors(
+    errors: FieldErrors<{
+      songName: string;
+      date: string;
+      select: string;
+      checkbox: boolean;
+      radio: null;
+      upload: null;
+    }>
+  ) {
+    const isTextWrong = errors.songName;
+    const isDateWrong = errors.date;
+    const isRadioWrong = errors.radio;
+
+    if (isTextWrong || isDateWrong || isRadioWrong) {
+      setpopUpClass("error-visible");
+      setpopUpText("You have to correct errors in form first only then you can submit.");
+    }
+  }
+
+  function handleData(data: {
+    songName: string;
+    date: string;
+    select: string;
+    checkbox: boolean;
+    radio: null;
+    upload: null | FileList;
+  }) {
+    let image = "";
+
+    try {
+      image = URL.createObjectURL(data.upload![0]);
+    } catch (error) {
+      image = "";
+    } finally {
+      const formData = [
+        data.songName,
+        data.date,
+        data.select,
+        data.checkbox.toString(),
+        data.radio as unknown as string,
```

Comment:
you should not use `as unknown as` [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions).

```suggestion
        data.radio || "",
```